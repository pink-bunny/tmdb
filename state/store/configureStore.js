import { createStore, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';

import reducer from './reducer';

const loggerMiddleware = (store) => (next) => (action) => {
  /* eslint-disable no-console */
  console.group(action.type);
  console.info('dispatching', action);
  const result = next(action);
  console.log('next state', store.getState());
  console.groupEnd();
  /* eslint-enable */

  return result;
};

export default function configureStore() {
  const middlewares = [loggerMiddleware, thunkMiddleware];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const store = createStore(reducer, undefined, composeWithDevTools(middlewareEnhancer));

  return store;
}
