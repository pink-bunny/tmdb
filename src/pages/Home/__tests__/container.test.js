import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';

import ConnectedHome from '../container';

jest.mock('../../../../state/session/selectors', () => ({
  isUserLoggedIn: jest.fn(() => true)
}));

describe('Header container', () => {
  const store = configureStore()({});
  const wrapper = shallow(<ConnectedHome store={store} />);
  const container = wrapper.dive().dive();

  it('should render dashboard', () => {
    expect(container).toMatchSnapshot();
  });

  it('should render login', () => {
    wrapper.setProps({ userLoggedIn: false, custom: 'Hey' });
    // console.log('SS', wrapper.props());

    expect(container).toMatchSnapshot();
  });
});
