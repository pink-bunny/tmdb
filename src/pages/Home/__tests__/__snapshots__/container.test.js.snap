// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header container. User is logged in. Should render dashboard 1`] = `<Connect(DashboardPage) />`;

exports[`Header container. User is not logged in. Should render login 1`] = `<Connect(WithFormik(Login)) />`;
